metrics/nop_reporter.go

```diff
@@ -99,3 +103,30 @@ type nopCachedTimer struct {
 
 func (nopCachedTimer) ReportTimer(interval time.Duration) {
 }
+
+// NopCachedHistogram is an implementation of tally.CachedHistogram
+var NopCachedHistogram tally.CachedHistogram = nopCachedHistogram{}
+
+type nopCachedHistogram struct {
```

Comment:
style nit: empty struct on a single line generally, right? or are we doing differerently:

```go
type nopCachedHistogram struct{}
```